feature_engine/encoding/similarity_encoder.py

```diff
@@ -198,20 +217,19 @@ def fit(self, X: pd.DataFrame, y: Optional[pd.Series] = None):
         X = check_X(X)
         self._check_or_select_variables(X)
         self._get_feature_names_in(X)
+        if self.keywords:
+            if not all(item in self.variables_ for item in self.keywords.keys()):
+                raise ValueError(
+                    "keywords have columns that are not present in the dataset"
+                )
         self.encoder_dict_ = {}
 
         if self.missing_values == "raise":
             _check_contains_na(X, self.variables_)
-            for var in self.variables_:
-                self.encoder_dict_[var] = (
-                    X[var]
-                    .astype(str)
-                    .value_counts()
-                    .head(self.top_categories)
-                    .index.tolist()
-                )
-        elif self.missing_values == "impute":
-            for var in self.variables_:
+        for var in self.variables_:
```

Comment:
are we removing the functionality to handle missing data? I guess, not because the tests pass. 

But regarding this code:
https://github.com/feature-engine/feature_engine/blob/a2c315db76cd6cd1327b1c7247cd797a63cec768/feature_engine/encoding/similarity_encoder.py#L236

Are we not making nan part of the categories to appear as dummies?